### PR TITLE
13 improve photo handling

### PR DESCRIPTION
**Pull request by davisshaver**
See https://github.com/fusioneng/shortcake-bakery/issues/13

screen shot 2015-07-06 at 2 01 55 pm

### CODE REVIEW

**Comment by davisshaver** [comment]
@danielbachhuber review/merge?

File: inc/shortcodes/class-facebook.php [hunk 1]
```diff
@@ -89,6 +89,7 @@ public static function callback( $attrs, $content = '' ) {
 			'#https?://(www)?\.facebook\.com\/video\.php\?v=[\d]+#',
 			'#https?:\/\/www?\.facebook\.com\/+.*?\/videos\/[\d]+\/#',
 			'#https?://(www)?\.facebook\.com\/permalink\.php\?story_fbid=[\d]+&id=[\d]+#',
+			'#https?:\/\/www?\.facebook\.com\/+.*?\/photos\/a\.[\d]+\.[\d]+\.[\d]+\/([\d])+/#',
```
Comment (danielbachhuber) [inline inc/shortcodes/class-facebook.php]: Will it always be `a.[0-9].[0-9].[0-9]`, or should we accommodate an arbitrary number of periods?

**Comment by danielbachhuber** [comment]
It'd be good to verify the potential photo URL patterns we'll need to support.

File: inc/shortcodes/class-facebook.php (hunk 1 above)
Comment (davisshaver) [inline inc/shortcodes/class-facebook.php]: @danielbachhuber yeah, that middle bit changes depending on album/position. I'm going with a more general regex to fast forward to the next forward slash and grab the digits after that.

**Comment by davisshaver** [comment]
@danielbachhuber as far as I can tell, `photos` is the new permalink for public photos.

File: inc/shortcodes/class-facebook.php [hunk 2]
```diff
@@ -89,6 +89,7 @@ public static function callback( $attrs, $content = '' ) {
 			'#https?://(www)?\.facebook\.com\/video\.php\?v=[\d]+#',
 			'#https?:\/\/www?\.facebook\.com\/+.*?\/videos\/[\d]+\/#',
 			'#https?://(www)?\.facebook\.com\/permalink\.php\?story_fbid=[\d]+&id=[\d]+#',
+			'#https?:\/\/www?\.facebook\.com\/+.*?\/photos\/([^/]+)/+([\d])+/#',
```
Comment (danielbachhuber) [inline inc/shortcodes/class-facebook.php]: Why would there be multiple forward-slashes? `/+`

Comment (davisshaver) [inline inc/shortcodes/class-facebook.php]: @danielbachhuber Oh. I thought that was a string operator...

**Comment by davisshaver** [comment]
@danielbachhuber Closed the loop with reversal test. Looks to be all good.

**Pull request merged**